src/codeflare_sdk/cluster/cluster.py

```diff
@@ -337,6 +337,67 @@ def get_current_namespace():  # pragma: no cover
         return "default"
 
 
+def get_cluster(cluster_name: str, namespace: str = "default"):
+    try:
+        config.load_kube_config()
+        api_instance = client.CustomObjectsApi()
+        rcs = api_instance.list_namespaced_custom_object(
+            group="ray.io",
+            version="v1alpha1",
+            namespace=namespace,
+            plural="rayclusters",
+        )
+    except Exception as e:
+        return _kube_api_error_handling(e)
+
+    for rc in rcs["items"]:
+        if rc["metadata"]["name"] == cluster_name:
+            machine_types = (
+                rc["metadata"]["labels"]["orderedinstance"].split("_")
+                if "orderedinstance" in rc["metadata"]["labels"]
+                else []
+            )
+            local_interactive = (
+                "volumeMounts"
+                in rc["spec"]["workerGroupSpecs"][0]["template"]["spec"]["containers"][
+                    0
+                ]
+            )
+            cluster_config = ClusterConfiguration(
+                name=rc["metadata"]["name"],
+                namespace=rc["metadata"]["namespace"],
+                machine_types=machine_types,
+                min_cpus=rc["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+                    "containers"
+                ][0]["resources"]["requests"]["cpu"],
+                max_cpus=rc["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+                    "containers"
+                ][0]["resources"]["limits"]["cpu"],
+                min_memory=int(
+                    rc["spec"]["workerGroupSpecs"][0]["template"]["spec"]["containers"][
+                        0
+                    ]["resources"]["requests"]["memory"][:-1]
+                ),
+                max_memory=int(
+                    rc["spec"]["workerGroupSpecs"][0]["template"]["spec"]["containers"][
+                        0
+                    ]["resources"]["limits"]["memory"][:-1]
+                ),
+                gpu=rc["spec"]["workerGroupSpecs"][0]["template"]["spec"]["containers"][
+                    0
+                ]["resources"]["limits"]["nvidia.com/gpu"],
+                instascale=True if machine_types else False,
+                image=rc["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+                    "containers"
+                ][0]["image"],
+                local_interactive=local_interactive,
+            )
+            return Cluster(cluster_config)
```

Comment:
This should be broken out into it's own function in order to improve readability.

Something like:
```python
def k8_cluster_obj_2_sdk_cluster_obj(rc):
```

Reply:
maybe even a function in `Cluster` like `from_k8_cluster_object(rc)`